catalyst-gateway/bin/src/service/api/cardano/staking/assets_get.rs

```diff
@@ -52,16 +56,30 @@ pub(crate) type AllResponses = WithErrorResponses<Responses>;
 /// # GET `/staked_ada`
 #[allow(clippy::unused_async, clippy::no_effect_underscore_binding)]
 pub(crate) async fn endpoint(
-    stake_address: Cip19StakeAddress, _provided_network: Option<Network>,
-    slot_num: Option<SlotNumber>,
+    stake_address: Cip19StakeAddress, _provided_network: Option<Network>, slot_num: Option<SlotNo>,
 ) -> AllResponses {
-    let persistent_res = calculate_stake_info(true, stake_address.clone(), slot_num).await;
+    let Some(persistent_session) = CassandraSession::get(true) else {
+        tracing::error!("Failed to acquire persistent db session");
+        return AllResponses::service_unavailable(
+            &anyhow::anyhow!("Failed to acquire db session"),
+            RetryAfterOption::Default,
+        );
+    };
+    let Some(volatile_session) = CassandraSession::get(false) else {
+        tracing::error!("Failed to acquire volatile db session");
+        return AllResponses::service_unavailable(
+            &anyhow::anyhow!("Failed to acquire volatile db session"),
+            RetryAfterOption::Default,
+        );
+    };
+    let persistent_res =
+        calculate_stake_info(&persistent_session, stake_address.clone(), slot_num).await;
```

Comment:
These should run in parallel.  ie, start them both and await the results from both.
All queries which have a volatile/persistent DB component should do the same, run in parallel.